Sources/GRPC/ConnectionManager.swift

```diff
@@ -176,38 +176,64 @@ internal final class ConnectionManager {
     }
   }
 
+  /// The last 'external' state we are in, a subset of the internal state.
+  private var externalState: ConnectivityState = .idle
+
+  /// Update the external state, potentially notifying a delegate about the change.
+  private func updateExternalState(to nextState: ConnectivityState) {
+    if self.externalState != nextState {
+      self._onConnectivityStateChange?(self, self.externalState, nextState)
+      self.externalState = nextState
+    }
+  }
+
+  /// Our current state.
   private var state: State {
     didSet {
       switch self.state {
       case .idle:
-        self.monitor.updateState(to: .idle, logger: self.logger)
+        self.updateExternalState(to: .idle)
         self.updateConnectionID()
 
       case .connecting:
-        self.monitor.updateState(to: .connecting, logger: self.logger)
+        self.updateExternalState(to: .connecting)
 
       // This is an internal state.
       case .active:
         ()
 
       case .ready:
-        self.monitor.updateState(to: .ready, logger: self.logger)
+        self.updateExternalState(to: .ready)
 
       case .transientFailure:
-        self.monitor.updateState(to: .transientFailure, logger: self.logger)
+        self.updateExternalState(to: .transientFailure)
         self.updateConnectionID()
 
       case .shutdown:
-        self.monitor.updateState(to: .shutdown, logger: self.logger)
+        self.updateExternalState(to: .shutdown)
       }
     }
   }
 
   /// The `EventLoop` that the managed connection will run on.
   internal let eventLoop: EventLoop
 
-  /// A connectivity state monitor.
-  internal let monitor: ConnectivityStateMonitor
+  /// A closure which is notified when the external connectivity state changes.
+  private var _onConnectivityStateChange: OnConnectivityStateChange?
+  internal typealias OnConnectivityStateChange =
+    (ConnectionManager, ConnectivityState, ConnectivityState) -> Void
+
+  /// A closure which is notified when the connection begins quiescing.
+  private var _onConnectionIsQuiescing: OnConnectionIsQuiescing?
+  internal typealias OnConnectionIsQuiescing = (ConnectionManager) -> Void
+
+  /// A closure which is notified when an HTTP/2 stream is closed.
+  private var _onHTTP2StreamClosed: OnHTTP2StreamClosed?
+  internal typealias OnHTTP2StreamClosed = (ConnectionManager) -> Void
+
+  /// A closure which is notified when the value of HTTP/2 SETTINGS_MAX_CONCURRENT_STREAMS changes.
+  private var _onHTTP2MaxConcurrentStreamsChanged: OnHTTP2MaxConcurrentStreamsChanged?
+  internal typealias OnHTTP2MaxConcurrentStreamsChanged = (ConnectionManager, Int) -> Void
```

Comment:
With a protocol this could be just plain functions. No `typealias` needed. IIRC `typealias` on functions doesn't play nicely with code completion.

Reply:
I didn't have any issues with the `typealias` with the Xcode I'm using. Either way it's protocol based now.